### ISSUE TITLE
Segmentation error-- Only crashes on Linux (OSX works)

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#464]
input data:
~~~~
sct_issues/20150820_issue_464/data
~~~~

~~~~
isct_orientation3d -i fmri_moco.nii -get
Segmentation fault
~~~~

~~~~
isct_orientation3d -i fmri.nii.gz -get
Input image orientation : RAI
~~~~

**Comment by jcohenadad** [comment]
strange thing: when specifying .nii extension, it works:
~~~~
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ ll
total 160
-rw-r--r-- 1 neuropoly neuropoly  16268 Aug 20 12:44 b0_moco_mean.nii.gz
-rw-r--r-- 1 neuropoly neuropoly 107345 Aug 20 12:44 dmri_moco.nii.gz
-rw-r--r-- 1 neuropoly neuropoly  28897 Aug 20 12:44 dwi_moco_mean.nii.gz
drwxr-xr-x 5 neuropoly neuropoly   4096 Aug 20 12:44 tmp.150820124356
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ isct_orientation3d -i dmri_moco.nii -get
Input image orientation : RAI
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ isct_orientation3d -i dmri_moco.nii.gz -get
Segmentation fault
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ 
~~~~

**Comment by jcohenadad** [comment]
strange thing: when specifying .nii extension, it works:
~~~~
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ ll
total 160
-rw-r--r-- 1 neuropoly neuropoly  16268 Aug 20 12:44 b0_moco_mean.nii.gz
-rw-r--r-- 1 neuropoly neuropoly 107345 Aug 20 12:44 dmri_moco.nii.gz
-rw-r--r-- 1 neuropoly neuropoly  28897 Aug 20 12:44 dwi_moco_mean.nii.gz
drwxr-xr-x 5 neuropoly neuropoly   4096 Aug 20 12:44 tmp.150820124356
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ isct_orientation3d -i dmri_moco.nii -get
Input image orientation : RAI
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ isct_orientation3d -i dmri_moco.nii.gz -get
Segmentation fault
neuropoly@ferguson:~/code/spinalcordtoolbox/testing/tmp.150820124356/results_sct_dmri_moco$ 
~~~~

**Comment by jcohenadad** [comment]
Possibly related with how the file is saved with nibabel. Here, it crashes in sct_fmri_moco:
~~~~
sct_fmri_moco:
sct.generate_output_file(path_tmp+'fmri'+param.suffix+'.nii', path_out+file_data+param.suffix+ext_data, param.verbose)

sct_utils:
convert(fname_in, fname_out)

sct_convert:
im = Image(fname_in)

msct_image/loadFromPath:
self.orientation = get_orientation(path)

sct_orientation/get_orientation:
status, output = sct.run('isct_orientation3d -i '+fname+' -get ', 1)

**Comment by benjamindeleener** [comment]
Tried on Ferguson.
Works on branch dev but not on branch jca_FSL_dependences

So probably related to image saving or to filename?

**Comment by benjamindeleener** [comment]
It is not related to filename. I created a temp folder tmp.test_folder and tried to get orientation from an image in it and it worked.

**Comment by benjamindeleener** [comment]
No differences between zipped and unzipped file (still in sct_test_data/data/fmri/)

**Comment by benjamindeleener** [comment]
When separating 4d image with fslsplit, it works:
```
fslsplit fmri_moco.nii.gz im -t
isct_orientation3d -i im0001.nii.gz -get
```
Output:
```
Input image orientation : RPI
```

**Comment by benjamindeleener** [comment]
In ITK mode:
the issue appears when reading the file.

**Comment by benjamindeleener** [comment]
One difference between fmri.nii and fmri_moco.nii.gz is the pixeltype:
fmri (works): short
fmri_moco (does not work): float

**Comment by benjamindeleener** [comment]
The issue is probably coming from a wrong image saving. But I fixed the issue by changing isct_orientation3d. The segmentation fault appeared when reading the image. However, it is not necessary to read the image to change its orientation. So I just removed the image reading and the issue is fixed.

**Comment by benjamindeleener** [comment]
It is not fixed...
For getting the proper orientation, the image needs to be read. Either way, it will return RAI orientation.

**Comment by benjamindeleener** [comment]
Fixed in commit 89df125d7d3a672c23edcf5de3b6f0191087a214

**Comment by jcohenadad** [comment]
can we close it?
